packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -231,6 +231,48 @@ function dedupeTransactions(txs: Transfer[]) {
   )
 }
 
+function getChainPairForTransfer(tx: Transfer): ChainPair | undefined {
+  if (isTransferTeleportFromSubgraph(tx)) {
+    // can't determine child chain id without fetching extra details
+    return undefined
+  }
+
+  // just in case local storage has incomplete details
+  if (!tx.parentChainId || !tx.childChainId) {
+    return undefined
+  }
+
+  return {
+    parentChainId: tx.parentChainId,
+    childChainId: tx.childChainId
+  }
+}
+
+function addToPrevFailedChainPairs(
```

Comment:
The logic is taken from the existing code (the block removed below), just reused between two hooks.